workerapi/worker.thrift

```diff
@@ -0,0 +1,38 @@
+enum Status {
+  UNKNOWN = 0
+  READY = 1
+  PENDING = 2
+  FAILURE = 3
+  RUNNING = 4
+  SUCCESS = 5
+  ABORTED = 6
+  TIMEOUT = 7
+}
+
+struct WorkerStatus {
+  1: required Status status
+  2: optional string taskId
+  3: optional string outUri
+  4: optional string errUri
+  5: optional string info
+  6: optional i32 exitCode
+}
+
+struct Task {
+  1: required list<string> argv
+  2: optional list<string> env
+  3: optional string dir
+  4: optional string taskId
+  5: optional string sourceId //Allows snapshotId or patchId
+  6: optional i32 timeout_ms
+}
+
+service Worker {
+  WorkerStatus QueryStatus()
+  WorkerStatus Run(1: Task task)
+  WorkerStatus Abort(
```

Comment:
Sure, the motivating case is that it lets us increase utilization because not all processes are created equal.

I think it *also* leads to a cleaner API that is less stateful. With the current API, if a Task gets scheduled but I didn't collect the results, whoops, I'm out of luck.

My suggestion would be:
Run(TaskDescription) -> ID (or maybe the whole Status)
Query(ID) -> Status
Abort(ID) -> Status

BUT, this API means that if you don't know what's running, you can't clear the Worker. Whoops.

One approach would be to have:
AbortAll()

BUT, that can be bad because you have a scheduler canceling jobs it doesn't know about.

What would be better is:
Running() -> List<ID>

Or, to get back to what you had that's pretty nice might be:
Status() -> struct WorkerStatus {
  List<ID>
  // other info that we add
}

So, WorkerStatus just says "I'm doing X, Y, Z". If the Scheduler wants to know about X, Y, Z, it has to query each. If it wants to kill X, Y, Z, it can do that explicitly.

IF there is another Scheduler that is scheduling A, B, C so that a Scheduler thought it had cleared the worker by killing X, Y, Z, then we can see the expectation violated explicitly, instead of having this "AbortAll" just confuse us.